scripts/map_blocks.py

```diff
@@ -0,0 +1,374 @@
+#!/usr/bin/env python3
+#
+# MAKE A BLOCK-ASSIGNMENT FILE FOR A SET OF SHAPES (E.G., DISTRICTS)
+#
+'''
+For example:
+
+./map_blocks.py district-shapes.geojson tl_2020_04_tabblock20.zip
+
+./map_blocks.py ~/dev/temp/AZ/district-shapes.geojson ~/dev/temp/AZ/tl_2020_04_tabblock20.zip
+./map_blocks.py ~/dev/temp/CA/district-shapes.geojson ~/dev/temp/CA/tl_2020_06_tabblock20.zip
+./map_blocks.py ~/dev/temp/TX/district-shapes.geojson ~/dev/temp/TX/tl_2020_48_tabblock20.zip
+./map_blocks.py ~/dev/temp/IL/Political_Township/Political_Township.geojson ~/dev/temp/IL/tl_2020_17_tabblock20.zip -s OBJECTID
+
+./map_blocks.py ~/dev/temp/AZ/district-shapes.geojson ~/dev/temp/AZ/tl_2020_04_tabblock20.zip -v
+./map_blocks.py ~/dev/temp/AZ/district-shapes.geojson ~/dev/temp/AZ/tl_2020_04_tabblock20.zip -b GEOID10 -s foo -v
+
+Bad inputs cases:
+
+./map_blocks.py district-shapes.json tl_2020_04_tabblock20.zip
+./map_blocks.py district-shapes.geojson tl_2020_04_tabblock20.shp
+./map_blocks.py ~/dev/temp/doesnt-exist.geojson ~/dev/temp/tl_2020_04_tabblock20.zip
+./map_blocks.py ~/dev/temp/district-shapes.geojson ~/dev/temp/doesnt-exist.zip
+
+For documentation, type:
+
+./map_blocks.py -h
+
+Setup instructions on Medium @ https://medium.com/dra-2020/making-a-block-assignment-file-a7fb7a68b9a
+
+'''
+
+import sys
+import os
+import argparse
+import csv
+
+# import pandas as pd
+import geopandas as gpd
+from tqdm import tqdm
+import json
+
+
+def main():
+    # Parse & validate the args
+
+    parser = argparse.ArgumentParser(description='Make a block-assignment file')
+    parser.add_argument('shapes_path', help='Path to .geojson for shapes being mapped, e.g., districts')
+    parser.add_argument('blocks_path', help='Path to .zip for block shapes being mapped to, e.g., 2020')
+    parser.add_argument('-b', '--blocks_key', dest='blocks_key', help='blocks key')
+    parser.add_argument('-s', '--shapes_key', dest='shapes_key', help='shape key')
+    parser.add_argument('-v', '--verbose', dest='verbose', action='store_true', help='verbose mode')
+
+    args = parser.parse_args()
+
+    shapes_path = os.path.abspath(args.shapes_path)
+    blocks_path = os.path.abspath(args.blocks_path)
+
+    verbose = args.verbose
+
+    shapes_fn, shapes_ext = os.path.splitext(shapes_path)
+    blocks_fn, blocks_ext = os.path.splitext(blocks_path)
+
+    if (shapes_ext != '.geojson'):
+        print("Container file must be a .geojson")
+        sys.exit(0)
+    
+    try:
+        f = open(shapes_path)
```

Comment:
You can text if a file/dir exists without opening like this:
   if os.path.exists(from_path)